frontend/src/metabase/parameters/components/ValuesSourceModal/ValuesSourceModal.unit.spec.tsx

```diff
@@ -255,16 +253,33 @@ describe("ValuesSourceModal", () => {
             name: "Products",
           }),
         ],
-        hasDataAccess: false,
+        hasCollectionAccess: false,
       });
 
       expect(
-        await screen.findByText("You don't have permissions to do that."),
+        screen.getByText("You don't have permissions to do that."),
+      ).toBeInTheDocument();
+    });
+
+    it("should allow searching for a card without access to the root collection (metabase#30355)", async () => {
```

Comment:
This test fails if we revert the change to `ValuesSourceCardModel` because it would try to load the root collection in this case